llvm/lib/MCCAS/MCCASObjectV1.cpp

```diff
@@ -1982,14 +1982,16 @@ struct DIEToCASConverter {
   bool IsLittleEndian;
   uint8_t AddressSize;
 
-  Error convertInNewDIEBlock(
-      DWARFDie DIE, DistinctDataWriter &DistinctWriter,
-      AbbrevSetWriter &AbbrevWriter,
-      SmallVectorImpl<std::unique_ptr<DIEDataWriter>> &DIEWriters);
+  struct ParentAndChildDIE {
+    DWARFDie Parent;
+    bool ParentAlreadyWritten;
+    unsigned WriterIndex;
```

Comment:
This should be just a `DIEWriter&`. Note that they are created with unique_ptrs, so their references will be stable.